0xPolygon/polygon-edge
# [Fix] Improve eth_gasPrice responce when --price-limit flag is set

# Description

Currently, `eth_gasPrice` does not take into account if `--price-limit` flag has been set by the chain operator.   
 Even if `--price-limit` flag is set to some value ( for example `100000` ), `eth_gasPrice` will return `0`.    

This PR fixes the issue. Now `eth_gasPrice` returns at minimum the value set by `--price-limit` flag.

# Changes include


# Checklist


## Testing


### Manual tests
Spin up a chain using this PR code, but run the server with `--price-limit` set to `100000` for example.    
Run curl targeting  `eth_gasPrice` endpoint. For example:     
`curl  localhost:8545 -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"eth_gasPrice","params":[],"id":1}'`     
     
You should get the value set in `--price-limit` flag ( `100000` ) from the API.

# Additional comments

Fixes EDGE-538
